llvm/lib/Analysis/MemoryBuiltins.cpp

```diff
@@ -706,7 +715,11 @@ SizeOffsetAPInt ObjectSizeOffsetVisitor::computeImpl(Value *V) {
   IntTyBits = DL.getIndexTypeSizeInBits(V->getType());
   Zero = APInt::getZero(IntTyBits);
 
+  APInt PrevConstantOffset = ConstantOffset;
+
+  ConstantOffset = ConstantOffset.sextOrTrunc(Offset.getBitWidth()) + Offset;
```

Comment:
Is it safe to possibly truncate here? The below bit with `IndexTypeSizeChanged` tries to very carefully handle the possibility that pointers (and hence indices on those pointers) have different sizes, it's a bit hard to see how exactly that should be handled here.

Reply:
I've moved to a call to `CheckedZextOrTrunc` which bails out in case of failure.